include/flatbuffers/flatbuffers.h

```diff
@@ -401,35 +401,153 @@ static inline bool StringLessThan(const char *a_data, uoffset_t a_size,
   return cmp == 0 ? a_size < b_size : cmp < 0;
 }
 
-struct String : public Vector<char> {
-  const char *c_str() const { return reinterpret_cast<const char *>(Data()); }
-  std::string str() const { return std::string(c_str(), size()); }
+template<typename T> struct StringBase : public T {
```

Comment:
This change I believe was only needed to support `char` related features. Please keeps `String` as it was.

Reply:
This may also be used to store a string in a byte array, but I can remove it if you wish.

Reply:
For now, please do. I'd like to see the impact of such changes separately.

Reply:
Removed.